File: structural/typescript/adapter_structured/adaptees.ts
export class AdvancedMediaPlayer {
  playVlc(fileName: string): void { console.log(`Playing vlc file: ${fileName}`); }
  playMp4(fileName: string): void { console.log(`Playing mp4 file: ${fileName}`); }
}

export class StripePaymentSystem {
  makeStripePayment(amount: number): void { console.log(`Processing $${amount} via Stripe`); }
}

export class PayPalPaymentSystem {
  sendPayPalRequest(amount: number): void { console.log(`Processing $${amount} via PayPal`); }
}

File: structural/dart/adapter_structured/adaptees.dart
class AdvancedMediaPlayer {
  void playVlc(String fileName) {
    print('Playing vlc file: $fileName');
  }
  
  void playMp4(String fileName) {
    print('Playing mp4 file: $fileName');
  }
}

class StripePaymentSystem {
  void makeStripePayment(double amount) {
    print('Processing \$$amount via Stripe');
  }
}

class PayPalPaymentSystem {
  void sendPayPalRequest(double amount) {
    print('Processing \$$amount via PayPal');
  }
}
